### PR TITLE
Added naive_coalesce to api2.

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

**Comment by tpoterba** [comment]
I've always disliked this being a separate method from `repartition`. We can discuss that later though, better to complete the API first!

File: python/hail/api2/matrixtable.py [hunk 1]
```diff
@@ -2091,6 +2091,33 @@ def repartition(self, num_partitions, shuffle=True):
...
+
+        Example
+        -------
+        Naively repartition to 10 partitions:
+
+        >>> dataset_result = dataset.naive_coalesce(10)
+
+        Warning
+        -------
+        :meth:`naive_coalesce` simply combines adjacent partitions to achieve the desired number.  It does not attempt to rebalance, unlike :meth:`repartition`, so it can produce a heavily unbalanced dataset.  An unbalanced dataset can be inefficient to operate on because the work is not evenly distributed across partitions.
```
Comment (tpoterba) [inline python/hail/api2/matrixtable.py]: Line too long. Use this tool to format (set spaces to 8, paste, hit enter)

https://hail.is/tools/formatter.html

File: python/hail/api2/matrixtable.py [hunk 2]
```diff
@@ -2091,6 +2091,33 @@ def repartition(self, num_partitions, shuffle=True):
...
+
+        Parameters
+        ----------
+        max_partitions : int
+            Desired number of partitions.  If the current number of partitions is less than or equal to ``max_partitions``, do nothing.
```
Comment (tpoterba) [inline python/hail/api2/matrixtable.py]: Line too long. Also, we're using the convention of referencing function parameters with italics (single backticks)

File: python/hail/api2/matrixtable.py [hunk 3]
```diff
@@ -2091,6 +2091,33 @@ def repartition(self, num_partitions, shuffle=True):
...
+
+        Returns
+        -------
+        :class:`MatrixTable`
+            Matrix table with the number of partitions equal to at most ``max_partitions``
```
Comment (tpoterba) [inline python/hail/api2/matrixtable.py]:
```
Matrix table with at most `max_partitions` partitions.
```

(italics instead of code, and period at the end)

**Comment by cseed** [comment]
I agree.  In fact, I've never liked repartition.  As we're using things, we basically never, ever want to do a shuffle and I'm not sure I've seen a case where the double-scan of repartition is worth the cost.

 > We can discuss that later though, better to complete the API first!

Agreed!

Good comments, will address.

File: python/hail/api2/tests.py [hunk 4]
```diff
@@ -431,6 +431,11 @@ def test_joins(self):
         self.assertTrue(rt.forall(rt.y2 == 2))
         self.assertTrue(ct.forall(ct.c2 == 2))
 
+    def test_naive_coalesce():
```
Comment (tpoterba) [inline python/hail/api2/tests.py]: needs self param

Comment (cseed) [reply python/hail/api2/tests.py]: Yep, just pushed a fix.

**Comment by cseed** [comment]
@tpoterba FYI, I added MatrixTable._same (I never thought that should be user-visible.) and set min_block_size=0 in the tests so we can load sample.vcf with more than one partition.  Let me know if you have objections.

**Comment by tpoterba** [comment]
Sounds fine to me

**Pull request merged**